public/components/wz-menu/wz-menu.js

```diff
@@ -47,349 +50,335 @@ import WzTextWithTooltipIfTruncated from '../../components/common/wz-text-with-t
+    }
 
-  getCurrentTab() {
-    const currentWindowLocation = window.location.hash;
-    let currentTab = '';
-    Object.keys(sections).some((section) => {
-      if (currentWindowLocation.match(`#/${section}`)) {
-        currentTab = sections[section];
-        return true;
-      }
-    });
-    return currentTab;
-  }
+    showToast = (color, title, text, time) => {
+      getToasts().add({
+        color: color,
+        title: title,
+        text: text,
+        toastLifeTimeMs: time,
+      });
+    };
 
-  loadApiList = async () => {
-    const result = await this.genericReq.request('GET', '/hosts/apis', {});
-    const APIlist = (result || {}).data || [];
-    if (APIlist.length) this.setState({ APIlist });
-  };
+    getCurrentTab() {
+      const currentWindowLocation = window.location.hash;
+      let currentTab = '';
+      Object.keys(sections).some((section) => {
+        if (currentWindowLocation.match(`#/${section}`)) {
+          currentTab = sections[section];
+          return true;
+        }
+      });
+      return currentTab;
+    }
 
-  loadIndexPatternsList = async () => {
-    try {
-      const list = await PatternHandler.getPatternList('api');
-      if (!list) return;
+    loadApiList = async () => {
+      const result = await this.genericReq.request('GET', '/hosts/apis', {});
+      const APIlist = (result || {}).data || [];
+      if (APIlist.length) this.setState({ APIlist });
+    };
 
-      // Abort if we have disabled the pattern selector
-      if (!AppState.getPatternSelector()) return;
+    loadIndexPatternsList = async () => {
+      try {
+        const list = await PatternHandler.getPatternList('api');
+        if (!list) return;
+
+        // Abort if we have disabled the pattern selector
+        if (!AppState.getPatternSelector()) return;
+
+        let filtered = false;
+        // If there is no current pattern, fetch it
+        if (!AppState.getCurrentPattern()) {
+          AppState.setCurrentPattern(list[0].id);
+        } else {
+          // Check if the current pattern cookie is valid
+          filtered = list.find((item) => item.id.includes(AppState.getCurrentPattern()));
+          if (!filtered) AppState.setCurrentPattern(list[0].id);
+        }
 
-      let filtered = false;
-      // If there is no current pattern, fetch it
-      if (!AppState.getCurrentPattern()) {
-        AppState.setCurrentPattern(list[0].id);
-      } else {
-        // Check if the current pattern cookie is valid
-        filtered = list.find(item =>
-          item.id.includes(AppState.getCurrentPattern())
-        );
-        if (!filtered) AppState.setCurrentPattern(list[0].id);
+        const data = filtered
+          ? filtered
+          : await this.indexPatterns.get(AppState.getCurrentPattern());
+        this.setState({ theresPattern: true, currentPattern: data.title });
+
+        // Getting the list of index patterns
+        if (list) {
+          this.setState({
+            patternList: list,
```

Comment:
**nitpick**: If we move the list of valid index patterns to the Redux store, we should not use the component state for this case. Take into account this for renders or conditional, but it should use the values from the store.

Reply:
Removed in https://github.com/wazuh/wazuh-kibana-app/pull/3598/commits/bff95d08d148140258ac93ea1255b739e88c5d71